packages/frontend/src/context/MyContext.tsx

```diff
@@ -102,6 +113,45 @@ export const MyContextProvider = (
     address: address as `0x${string}`,
   });
 
+  const getStores = async () => {
+    const stores = new Map();
+
+    const logs = await publicClient.getLogs({
+      address: abi.addresses.ShopReg as Address,
```

Comment:
we want to get only the event  `Transfer` so we need to add something like  `event: abi.Transfer`  That way we only get the transfer events. Transfer emits the following     `event Transfer(address indexed _from, address indexed _to, uint256 indexed _tokenId);`  so you can build the map of merchants to shops without anyother queries